File: Python/Problem_090.py
from itertools import combinations

def expand69(s: set[int]) -> set[int]:
    if 6 in s or 9 in s:
        s = set(s)
        s.add(6)
        s.add(9)
    return s

def euler_90() -> int:
    digits = range(10)
    cubes = [set(c) for c in combinations(digits, 6)]

    req = [(0,1),(0,4),(0,9),(1,6),(2,5),(3,6),(4,9),(6,4),(8,1)]

    count = 0
    for i in range(len(cubes)):
        A = expand69(cubes[i])
        for j in range(i, len(cubes)):
            B = expand69(cubes[j])

            ok = True
            for x, y in req:
                if not ((x in A and y in B) or (y in A and x in B)):
                    ok = False
                    break
            if ok:
                count += 1
    return count

if __name__ == "__main__":
    print(euler_90())  # 1217

File: Java/Problem_090.java
import java.util.*;

public class Euler90 {
    static void expand69(boolean[] b) {
        if (b[6] || b[9]) { b[6] = true; b[9] = true; }
    }

    static List<boolean[]> combinations() {
        List<boolean[]> res = new ArrayList<>();
        int[] d = {0,1,2,3,4,5,6,7,8,9};
        int[] idx = {0,1,2,3,4,5};

        while (true) {
            boolean[] b = new boolean[10];
            for (int k = 0; k < 6; k++) b[d[idx[k]]] = true;
            res.add(b);

            int i = 5;
            while (i >= 0 && idx[i] == 10 - 6 + i) i--;
            if (i < 0) break;
            idx[i]++;
            for (int j = i + 1; j < 6; j++) idx[j] = idx[j - 1] + 1;
        }
        return res;
    }

    public static void main(String[] args) {
        List<boolean[]> cubes = combinations();
        int[][] req = {
            {0,1},{0,4},{0,9},{1,6},{2,5},{3,6},{4,9},{6,4},{8,1}
        };

        long count = 0;
        int n = cubes.size();

        for (int i = 0; i < n; i++) {
            boolean[] A = cubes.get(i).clone();
            expand69(A);

            for (int j = i; j < n; j++) {
                boolean[] B = cubes.get(j).clone();
                expand69(B);

                boolean ok = true;
                for (int[] p : req) {
                    int x = p[0], y = p[1];
                    if (!((A[x] && B[y]) || (A[y] && B[x]))) {
                        ok = false;
                        break;
                    }
                }
                if (ok) count++;
            }
        }

        System.out.println(count); // 1217
    }
}
